### PR TITLE
feat: added wishlist management

### PR DESCRIPTION
**Pull request by ChhikaraBRUH**
Preview Link: https://deploy-preview-4--snkrs-store.netlify.app/

Added wishlist management:

-Added wishlist page functionality
-Added toggle add/remove from wishlist by clicking on heart icon on card
-Added number of wishlisted items to navbar
-Added login requirements for wishlisting
-Added Out Of Stock Styling to cards

x- Add to cart and cart management remaining

### CODE REVIEW

**Review by BISINIGISHIVANI — APPROVED** [review]
small suggestion:cart item you can add button save for later or move to wish list.
and all functionality working fine

File: src/components/ProductCard/ProductCard.css [hunk 1]
```diff
@@ -1,3 +1,35 @@
...
+	display: flex;
+	align-items: center;
+	justify-content: center;
+}
+
+.card-out-of-stock-img-div img {
+	filter: grayscale(85%);
+}
+
+.card-out-of-stock-text {
```
Comment (Jay-Gadhiya) [inline src/components/ProductCard/ProductCard.css:17]: good class names

**Comment by ChhikaraBRUH** [comment]
sure @BISINIGISHIVANI 
Will do next in the cart update.

File: src/pages/Wishlist/components/WishlistCard.jsx [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+	const { addToWishlist, removeFromWishlist, isItemWishlisted } = useWishlist();
+	const itemInWishlist = isItemWishlisted(product);
+
+	const wishlistToggle = async (product) => {
+		try {
+			if (!itemInWishlist) {
+				await addToWishlist(product);
+			} else {
+				await removeFromWishlist(product);
+			}
```
Comment (Jay-Gadhiya) [inline src/pages/Wishlist/components/WishlistCard.jsx:13]: you can use ternary operator

**Comment by Jay-Gadhiya** [comment]
-context and reducer coded well
- UI looks good
- wishlist functionality working fine

**Review by Jay-Gadhiya — APPROVED** [review]

**Comment by ChhikaraBRUH** [comment]
thank you. will add and improve code now with the cart functionality.

**Pull request merged**